qiskit/quantum_info/synthesis/one_qubit_decompose.py

```diff
@@ -81,45 +80,62 @@ def __call__(self, unitary_mat, simplify=True, atol=DEFAULT_ATOL):
                               "input matrix is not unitary.")
         circuit = self._circuit(unitary_mat, simplify=simplify, atol=atol)
         # Check circuit is correct
-        if not Operator(circuit).equiv(unitary_mat):
+        if not Operator(circuit).equiv(Operator(unitary_mat)):
             raise QiskitError("OneQubitEulerDecomposer: "
                               "synthesis failed within required accuracy.")
         return circuit
 
-    def _angles(self, unitary_mat, atol=DEFAULT_ATOL):
+    def _angles(self, unitary_mat):
         """Return Euler angles for given basis."""
-        if self._basis in ['U3', 'U1X', 'ZYZ', 'ZXZ']:
+        if self._basis in ['U3', 'U1X', 'ZYZ']:
             return self._angles_zyz(unitary_mat)
+        if self._basis == 'ZXZ':
+            return self._angles_zxz(unitary_mat)
         if self._basis == 'XYX':
-            return self._angles_xyx(unitary_mat, atol=atol)
+            return self._angles_xyx(unitary_mat)
         raise QiskitError("OneQubitEulerDecomposer: invalid basis")
 
     def _circuit(self, unitary_mat, simplify=True, atol=DEFAULT_ATOL):
-        # Add phase to matrix to make it special unitary
-        # This ensure that the quaternion representation is real
-        angles = self._angles(unitary_mat)
+        theta, phi, lam, phase = self._angles(unitary_mat)
```

Comment:
the `phase` is assigned here but never used within this function. is the idea that you'll later add phase correction using this info?

Reply:
The phase value isn't used in this PR but is still returned. It will be used to make 1Q synthesis phase-correct in the phase PR.